py-polars/polars/dataframe/frame.py

```diff
@@ -2286,7 +2283,40 @@ def to_pandas(  # noqa: D417
                 else:
                     raise ModuleNotFoundError(msg)
 
-        record_batches = self._df.to_pandas()
+        object_columns = [
+            name for (name, dtype) in zip(self.columns, self.dtypes) if dtype == Object
+        ]
+        object_columns_set = set(object_columns)
+
+        if len(object_columns) == len(self.columns):
```

Comment:
Using `width` is more efficient:

```suggestion
        if len(object_columns) == self.width:
```

Reply:
I decided to drop this fast path in the interest of clarity, but did equivalent change elsewhere. In practice it seems like it's rare that all columns will be `Object` so extra fast path is unnecessary.